src/test/scala/com/lightbend/kafkalagexporter/LookupTableSpec.scala

```diff
@@ -4,16 +4,245 @@
 
 package com.lightbend.kafkalagexporter
 
-import com.lightbend.kafkalagexporter.LookupTable.Table.{LagIsZero, Prediction, TooFewPoints}
+import java.time.{Clock, Instant, ZoneId}
+
+import com.lightbend.kafkalagexporter.LookupTable.Table._
+import com.redis.RedisClient
+
 import org.scalatest.freespec.AnyFreeSpec
 import org.scalatest.matchers.should.Matchers
 
+import scala.concurrent.duration.{DurationInt, DurationLong}
+
 class LookupTableSpec extends AnyFreeSpec with Matchers {
 
   import com.lightbend.kafkalagexporter.LookupTable._
 
   "LookupTable" - {
-    "lookupOffset" - {
+
+    "toLong" in {
+      toLong("1646985399302") shouldBe Some(1646985399302L)
+      toLong("string") shouldBe None
+    }
+
+    "RedisTable" - {
```

Comment:
I would prefer not to have duplication of these test cases. When you introduce the lookup table base type you can setup each implementation, add them to a list, and then run all of these tests for each.

i.e.

```
lookupTableImpls.foreach { table =>
  "invalids and edge conditions" in {
    if (table.length(redisClient) > 0) {
      fail(s"New table should be empty $table")
    }
    ...
  } 
}
```